### PR TITLE
Object Replication - List Blobs Item Properties

### PR DESCRIPTION
**Pull request by amnguye**
See/Related to PR #11909 
Object Replication for List Blobs Item Properties

This PR enables users to retrieve a dictionary that contains the details of the Object Replication Metadata (OrMetadata) applied to the source blob when calling List/Get Blobs. The dictionary is formatted as so Dictionary< policy_id, Dictionary < rule_id, status of replication(Complete,Failed) >(e.g. { policy_id1, { {rule_id1, Complete}, {rule_id2, Complete} }; {policy_id2, { {rule_id3, Complete} };..).

Also comments were updated from the last PR to better help understand how the Dictionary is formatted.

TODO: There's a bug in the generator where if the XML name is set in the main properties object, it will default taking the name of the object as the XML tag to look for, instead of looking at the reference object where it has the XML name defined. There's a transform in this PR to work around this for now.

### CODE REVIEW

**Review by seanmcc-msft — APPROVED** [review]

**Comment by amnguye** [comment]
We will wait to merge this into master after the stg73base merges into master. The next PR will have the APIview comments in #12740 .

**Pull request closed without merging**
